tests/load-tests/cluster_read_config.yaml

```
@@ -77,3 +77,45 @@
 - name: measurements.node_disk_io_time_seconds_total
   monitoring_query: sum(irate(node_disk_io_time_seconds_total{cluster="",namespace=~".*"}[5m]))
   monitoring_step: 15
+
+# Interesting CI environment variables
+{% for var in [
+  'BUILD_ID',
+  'HOSTNAME',
+  'JOB_NAME',
+  'OPENSHIFT_API',
+  'PROW_JOB_ID',
+  'PULL_BASE_REF',
+  'PULL_BASE_SHA',
+  'PULL_HEAD_REF',
+  'PULL_NUMBER',
+  'PULL_PULL_SHA',
+  'PULL_REFS',
+  'REPO_NAME',
+  'REPO_OWNER',
+  'SCENARIO',
+] %}
+- name: metadata.env.{{ var }}
+  env_variable: {{ var }}
+{% endfor %}
+
+# Cluster nodes info
+- name: metadata.cluster.control-plane.count
+  command: oc get nodes -l node-role.kubernetes.io/master -o name | wc -l
+
+- name: metadata.cluster.control-plane.flavor
+  command: oc get nodes -l node-role.kubernetes.io/master -o json | jq --raw-output '.items | map(.metadata.labels."beta.kubernetes.io/instance-type") | unique | sort | join(",")'
+
+- name: metadata.cluster.control-plane.nodes
+  command: oc get nodes -l node-role.kubernetes.io/master -o json | jq '.items | map(.metadata.name)'
+  output: json
+
+- name: metadata.cluster.compute-nodes.count
+  command: oc get nodes -l node-role.kubernetes.io/worker -o name | wc -l
+
+- name: metadata.cluster.compute-nodes.flavor
+  command: oc get nodes -l node-role.kubernetes.io/worker -o json | jq --raw-output '.items | map(.metadata.labels."beta.kubernetes.io/instance-type") | unique | sort | join(",")'
+
+- name: metadata.cluster.compute-nodes.nodes
+  command: oc get nodes -l node-role.kubernetes.io/master -o json | jq '.items | map(.metadata.name)'
+  output: json
```

```suggestion
  output: json

- name: metadata.scenario-content
  command: cat /usr/local/ci-secrets/redhat-appstudio-load-test/load-test-scenario.${SCENARIO}
```

@jhutar Could you also add the above? That will add the actual scenario content (the env variables exported by the scenario).

---

I couldn't come up with better name then `scenarioContent` so please feel free to change it :) maybe `scenarioEnv` or just simply `scenario`

---

In the JSON it would be like:
```
        ...        
        "PULL_REFS": "main:507f69fa2efb3ed40271d5cc8436580265e24255,761:a95112da6def2f1354e982ec53423cb5674a6838",
        "SCENARIO": "ci-10u-10t-nodejs"
    },
    "scenarioContent": "export USER_PREFIX=ci-nodejs USERS_PER_THREAD=10 THREADS=10 COMPONENT_REPO=https://github.com/nodeshift-starters/devfile-sample.git KUBE_SCHEDULER_LOG_LEVEL=TraceAll"
},
```

---

Cool, thanks. Need to tweak the variable. I think it is not expanded this way. Mnt.

---

Nope, I take it back, it works, sorry :)